includes/rest-api/class-sensei-rest-api-lesson-quiz-controller.php

```diff
@@ -230,6 +230,16 @@ public function save_quiz( WP_REST_Request $request ) {
 
 		Sensei()->quiz->set_questions( $quiz_id, array_filter( $question_ids ) );
 
+		/**
+		 * Fire action when a the lesson-quiz controller updates the data of the quiz.
+		 *
+		 * @hook sensei_rest_api_lesson_quiz_update
+		 *
+		 * @param {WP_Post} $lesson         The lesson.
+		 * @param {array}   $request_params The request parameters.
+		 */
+		do_action( 'sensei_rest_api_lesson_quiz_update', $lesson, $json_params );
```

Comment:
Wouldn't it be better if it was a **filter** hook, instead of action? The filter hook could allow modification of the quiz data before saving it.

Reply:
Hmm yeah I thought about this too but the thing is that the `save_quiz` method updates/creates also questions and the lesson. If we use a filter we will need to add it before the quiz `wp_insert_post` which can lead to 2 issues:
1. When the filter will be applied questions and the meta in the lesson might not exist which will be error prone if we won't to update the questions.
2. The filter won't simply filter the data and it will have side effect If we won't to update posts other than the quiz.

Reply:
Ohhh. I get it now. Thanks for clearing that out. 👍